inception/embedding_service.py

```diff
@@ -133,30 +134,44 @@ async def generate_query_embedding(self, text: str) -> list[float]:
         return embedding[0].tolist()
 
     async def generate_text_embeddings(
-        self, texts: list[str]
-    ) -> list[list[ChunkEmbedding]]:
-        """Generate embeddings for a list of texts"""
+        self, texts: dict[int, str]
+    ) -> list[TextResponse]:
+        """Generate embeddings for a dict of texts"""
         if not texts:
-            raise ValueError("Empty text list")
+            raise ValueError("Empty text dict")
 
-        all_embeddings = []
-        all_chunks = []
-        chunk_counts = []
+        logger.info(f"Generating embedding for {len(texts)} documents")
 
-        logger.info(
-            f"Generating embedding for {len(texts)} documents of {sum(len(s) for s in texts)} characters"
-        )
         start_time = time.time()
 
-        # Collect chunks
-        for text in texts:
-            chunks = self.split_text_into_chunks(text)
-            CHUNK_COUNT.labels(endpoint="text").inc(len(chunks))
-            all_chunks.extend(chunks)
-            chunk_counts.append(len(chunks))
+        # Chunk the texts
+        all_chunks = []
+        chunk_counts = []
+        chunks_by_id = {}
+
+        with ThreadPoolExecutor() as executor:
```

Comment:
By default, the `ThreadPoolExecutor` sets `max_workers` to cpu_count + 4, capped at 32.

If the chunking process is completely CPU-bound, adjusting this value might not help much. However, it might be worth making `max_workers` configurable via a setting, so we can run performance tests in production or try setting it to a lower value? like `os.cpu_count() - 1 or  1`